shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -327,7 +333,17 @@ abstract class JSBackend(allowUnresolvedSymbols: Bool) {
         case R(b) => CaseOf(desugarQuote(trm), b)
       }
     case _ if term.desugaredTerm.isDefined => desugarQuote(term.desugaredTerm.getOrElse(die))
-    case _ => term // * For other terms, either they are not supported & there would be a type error, or we don't need desugar them
+    case Assign(lhs, rhs) if !isQuoted => Assign(desugarQuote(lhs), desugarQuote(rhs))
+    case NuNew(cls) if !isQuoted => NuNew(desugarQuote(cls))
+    case TyApp(lhs, targs) if !isQuoted => TyApp(desugarQuote(lhs), targs)
+    case Forall(p, body) if !isQuoted => Forall(p, desugarQuote(body))
+    case Inst(body) if !isQuoted => Inst(desugarQuote(body))
+    case _: Super if !isQuoted => term
+    case Eqn(lhs, rhs) if !isQuoted => Eqn(lhs, desugarQuote(rhs))
+    case While(cond, body) if !isQuoted => While(desugarQuote(cond), desugarQuote(body))
+    case _: Bind | _: Test | If(_, _)  | _: Splc | _: Where | _: AdtMatchWith | _: Rft | _: New
+        | _: Assign | _: NuNew | _: TyApp | _: Forall | _: Inst | _: Super | _: Eqn | _: While =>
```

Comment:
I think `Assign`, `While`, and `NuNew` would be easy enough to support here?

Reply:
They should be. But I still prefer to implement them in the next PR since this one is huge enough.

Reply:
Ok sure.